### PR TITLE
Tolerate Flavor information for computing instance type

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
Current devstack seems to return "id", and an upcoming change using
nova's microversion will be returning "original_name":
https://blueprints.launchpad.net/nova/+spec/instance-flavor-api

So let's just inspect what is present and use that to figure out
the instance type.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by NickrenREN** [comment]
a bit soon
/lgtm

File: pkg/cloudprovider/providers/openstack/openstack_instances.go [hunk 1]
```diff
@@ -188,19 +188,17 @@ func (i *Instances) InstanceType(name types.NodeName) (string, error) {
...
-		return "", fmt.Errorf("flavor name not present in server info")
-	}
-
-	flavor, ok := val.(string)
-
-	if !ok {
-		return "", fmt.Errorf("flavor name is not a string")
+	keys := []string{"name", "id", "original_name"}
+	for _, key := range keys {
+		val, found := srv.Flavor[key]
```
Comment (anguslees) [inline pkg/cloudprovider/providers/openstack/openstack_instances.go]: Minor style suggestion:  It's ever-so-slightly-more-idiomatic to inline trivial checks like these:

```go
if val, found := srv.Flavor[key]; found {
   if flavor, ok := val.(string); ok {
      return flavor, nil
   }
}
```

Comment (dims) [inline pkg/cloudprovider/providers/openstack/openstack_instances.go]: Ack @anguslees will fix it up

**Pull request merged**